src/Tty-macos.zig

```diff
@@ -0,0 +1,313 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const os = std.os;
+const Vaxis = @import("vaxis.zig").Vaxis;
+const Parser = @import("Parser.zig");
+const GraphemeCache = @import("GraphemeCache.zig");
+const select = @import("select.zig").select;
+
+const log = std.log.scoped(.tty);
+
+const Tty = @This();
+
+const Writer = std.io.Writer(os.fd_t, os.WriteError, os.write);
+const BufferedWriter = std.io.BufferedWriter(4096, Writer);
+
+/// the original state of the terminal, prior to calling makeRaw
+termios: os.termios,
+
+/// the file descriptor we are using for I/O
+fd: std.fs.File,
+
+/// the write end of a pipe to signal the tty should exit its run loop
+quit_fd: ?os.fd_t = null,
+
+buffered_writer: BufferedWriter,
+
+/// initializes a Tty instance by opening /dev/tty and "making it raw"
+pub fn init() !Tty {
+    // Open our tty
+        const fd = try std.fs.cwd().openFile("/dev/tty", .{
+            .mode = .read_write,
+            .allow_ctty = true,
+});
+
+    // Set the termios of the tty
+    const termios = try makeRaw(fd.handle);
+
+    return Tty{
+        .fd = fd,
```

Comment:
It looked like in the `select` function you were using the `fd_t` of this file, is there a reason you kept this type as the `fs.File` instead of just the handle?

Reply:
not really, happy to just use the `fd_t`!